controllers/openstacknet_controller.go

```diff
@@ -192,7 +182,39 @@ func (r *OpenStackNetReconciler) Reconcile(ctx context.Context, req ctrl.Request
 			return ctrl.Result{}, err
 		}
 
+		// get bridgeName from desiredState
 		bridgeName := gjson.Get(string(desiredState["desiredState"]), "interfaces.#.name").Array()[0]
+
+		// Don't add a uniq  label here because the NCP might be related to multiple networks/NetworkAttachmentDefinitions
+		labelSelector := map[string]string{
+			common.OwnerNameSpaceLabelSelector:      instance.Namespace,
+			common.OwnerControllerNameLabelSelector: openstacknet.AppLabel,
+		}
+
+		// generate NodeNetworkConfigurationPolicy for the bridge
+		ncp := common.NetworkConfigurationPolicy{
+			Name:                           bridgeName.String(),
+			Labels:                         labelSelector,
+			NodeNetworkConfigurationPolicy: instance.Spec.AttachConfiguration.NodeNetworkConfigurationPolicy,
+		}
+		err = common.CreateOrUpdateNetworkConfigurationPolicy(r, instance, instance.Kind, &ncp)
+		if err != nil {
+			instance.Status.CurrentState = ospdirectorv1beta1.NetError
+			_ = r.setStatus(instance, oldStatus, err.Error())
+			return ctrl.Result{}, err
+		}
+
+		// TODO: add finalizer to the NCP to prevent the bridge from being deleted if a network gets deleted
```

Comment:
Could another option be to have this `OSNet` finalizer get all other `OSNet`s in the namespace and then only delete the `NNCP` if no other `OSNet`s are using the bridge?

Reply:
yes, thats another possible implementation